src/Controller/PricingAndPlansController.php

```diff
@@ -107,63 +130,53 @@ public function catalogPage(UserInterface $user) {
-    return $this->controller_factory->ratePlanController($package_id);
+    // Load rate plans for each package.
+    foreach (Package::getAvailableApiPackagesByDeveloper($user->getEmail()) as $package) {
+      /** @var \Drupal\apigee_m10n\Entity\PackageInterface $package */
+      foreach ($package->get('ratePlans') as $rate_plan) {
+        $rate_plans["{$package->id()}:{$rate_plan->target_id}"] = $rate_plan->entity;
+      };
+    }
+
+    return $this->buildPage($rate_plans);
   }
 
   /**
-   * Call back to pre-process the entity list.
+   * Builds the page for a rate plan listing.
    *
-   * @param array $build
-   *   A render array as processed by
-   *   `\Drupal\Core\Entity\EntityViewBuilder::buildMultiple()`.
+   * @param \Drupal\apigee_m10n\Entity\RatePlanInterface[] $plans
+   *   A list of rate plans.
    *
    * @return array
-   *   The processed render array.
+   *   A render array for plans.
    */
-  public function preRender(array $build) {
-    // Add a wrapper around the list.
-    $build['#prefix'] = '<ul class="apigee-package-list">';
-    $build['#suffix'] = '</ul>';
-    // Wrap each package as a list item.
-    foreach (Element::children($build) as $index) {
-      // Header HTML.
-      $header_html = '
-        <div class="apigee-sdk-package-basic">
-          <div class="apigee-package-label">@package_label</div>
-          <div class="apigee-sdk-product-list-basic">(<span class="apigee-sdk-product"> @product_labels </span>)</div>
-        </div>
-        <div class="apigee-package-details">';
-      // Build a list of product labels for the basic info header.
-      $product_labels = [];
-      foreach ($build[$index]['#package']->get('apiProducts') as $item) {
-        $product_labels[] = $item->entity->label();
-      }
-      $build[$index]['#prefix'] = new FormattableMarkup('<li class="apigee-sdk-package">' . $header_html, [
-        '@package_label' => $build[$index]['#package']->label(),
-        '@product_labels' => implode(', ', $product_labels),
-      ]);
-      $build[$index]['#suffix'] = '</div></li>';
+  protected function buildPage($plans) {
+    $build = [
+      '#type' => 'container',
+      '#attributes' => ['class' => ['rate-plan-list']],
+      '#cache' => [
+        'contexts' => [
+          'user',
+          'url.developer',
+        ],
+        'tags' => [],
+        'max-age' => 300,
+      ],
+      '#attached' => ['library' => ['apigee_m10n/rate_plan.entity_list']],
+    ];
+
+    foreach ($plans as $plan) {
+      // TODO: Add a test for render cache.
+      $build['#cache']['tags'] = Cache::mergeTags($build['#cache']['tags'], $plan->getCacheTags());
     }
+    // Get the view mode from package config.
+    $view_mode = $this->config(RatePlanConfigForm::CONFIG_NAME)->get('catalog_view_mode');
+
+    // Generate a build array using the view builder.
+    $build['plan_list'] = $this->entityTypeManager()->getViewBuilder('rate_plan')->viewMultiple($plans, $view_mode ?? 'default');
```

Comment:
Can you please make this an item-list?  In order to control the layout of this I need to be able wrap these in column classes, and doing that on the entity view mode itself is a bad practice that may break stuff elsewhere for people in the future.

Reply:
BTW, will need `'#theme' => ['item_list__rate_plan_list'],` for this, so it can be targeted as well.

Reply:
Using automatic theme suggestions like `'#theme' => 'item_list__rate_plan_list'` or `'#theme' => 'container__rate_plan_list'` is a great idea but item list doesn't play well with the entity list builder because the entity list builder adds `#pre_render` and `#sorted` in which the former crashes the item-list template.

Reply:
What do you suggest I do for wrapping these in columns?   Can you make a theme function that allows me to wrap each row?

Reply:
We switched from using `view_builder->viewMultiple()` to `view_builder->view()` for each rate plan to make the list more compatible with containers/lists.